File: src/eqn/ast/EqnAstNodeExp.java
package eqn.ast;

import eqn.parser.exception.EqnException;
import org.jetbrains.annotations.NotNull;

public class EqnAstNodeExp extends EqnAstNodePreDefinedUnaryFunction {

    public EqnAstNodeExp(@NotNull EqnAstNode equationNode) throws EqnException {
        super("exp", equationNode);
    }

    @Override
    public double evaluate() throws EqnException {
        return Math.exp(operand().evaluate());
    }

}

File: src/eqn/ast/EqnAstNodeExp.kt
package eqn.ast

import eqn.ast.base.EqnAstNode
import eqn.ast.base.EqnAstNodeUnary
import eqn.parser.exception.EqnException
import kotlin.math.exp

class EqnAstNodeExp(operand: EqnAstNode) : EqnAstNodeUnary("exp", Type.PreDefinedUnaryFunction, PrecedenceType.Function, operand) {
    @Throws(EqnException::class)
    override fun evaluate(arguments: Map<String, Double>?): Double {
        return exp(operand.evaluate(arguments))
    }
}